californium-core/src/main/java/org/eclipse/californium/core/network/EndpointManager.java

```diff
@@ -81,235 +92,112 @@ public static EndpointManager getEndpointManager() {
-	 * To listen on specific interfaces or ports, set the default endpoint manually.
-	 * To distinguish different interfaces, one endpoint per interface must be added.
-	 */
-	private synchronized void createDefaultEndpoint() {
-		if (default_endpoint != null) return;
-
-		default_endpoint = new CoapEndpoint();
-
-		try {
-			default_endpoint.start();
-			LOGGER.log(Level.INFO, "Created implicit default endpoint {0}", default_endpoint.getAddress());
-		} catch (IOException e) {
-			LOGGER.log(Level.SEVERE, "Could not create default endpoint", e);
-		}
-	}
-
-	private synchronized void createTcpEndpoint() {
-		if (default_tcp_endpoint != null)
-			return;
-
-		NetworkConfig config = NetworkConfig.getStandard();
-		TcpClientConnector connector = new TcpClientConnector(config.getInt(NetworkConfig.Keys.TCP_WORKER_THREADS),
-				config.getInt(NetworkConfig.Keys.TCP_CONNECT_TIMEOUT),
-				config.getInt(NetworkConfig.Keys.TCP_CONNECTION_IDLE_TIMEOUT));
-
-		default_tcp_endpoint = new CoapEndpoint(connector, config);
-		try {
-			default_tcp_endpoint.start();
-			LOGGER.log(Level.INFO, "Created implicit tcp endpoint {0}", default_tcp_endpoint.getAddress());
-		} catch (IOException e) {
-			LOGGER.log(Level.SEVERE, "Could not create tcp endpoint", e);
-		}
-	}
-
-	private synchronized void createSecureTcpEndpoint() {
-		if (default_secure_tpc_endpoint != null)
-			return;
-
-		NetworkConfig config = NetworkConfig.getStandard();
-		TlsClientConnector connector = new TlsClientConnector(config.getInt(NetworkConfig.Keys.TCP_WORKER_THREADS),
-				config.getInt(NetworkConfig.Keys.TCP_CONNECT_TIMEOUT),
-				config.getInt(NetworkConfig.Keys.TCP_CONNECTION_IDLE_TIMEOUT));
-
-		default_secure_tpc_endpoint = new CoapEndpoint(connector, config);
-		try {
-			default_secure_tpc_endpoint.start();
-			LOGGER.log(Level.INFO, "Created implicit secure tcp endpoint {0}",
-					default_secure_tpc_endpoint.getAddress());
-		} catch (IOException e) {
-			LOGGER.log(Level.SEVERE, "Could not create secure tcp endpoint", e);
+		uriScheme = uriScheme.toLowerCase();
+		Endpoint endpoint = endpoints.get(uriScheme);
+		if (null == endpoint) {
+			endpoint = createDefaultEndpoint(uriScheme);
 		}
+		return endpoint;
 	}
 
 	/**
-	 * Configures a new secure tcp endpoint to use by default. Any old tcp endpoint is destroyed.
-	 *
-	 * @param endpoint the new default secure tcp endpoint.
+	 * Set the default endpoint for the supported scheme of the endpoint. Start
+	 * the endpoint, if its not already started. Stop the old endpoint.
+	 * 
+	 * @param newEndpoint new default endpoint for the supported scheme.
+	 * @throws IllegalArgumentException, if the uri scheme is not supported or
+	 *             the endpoint is null.
 	 */
-	public synchronized void setTcpEndpoint(Endpoint endpoint) {
-		if (this.default_tcp_endpoint != null) {
-			this.default_tcp_endpoint.destroy();
+	public synchronized void setDefaultEndpoint(Endpoint newEndpoint) throws IllegalArgumentException {
```

Comment:
we do not declare `RuntimeException`s being thrown

However, you should throw a `NullPointerException` instead of an `IllegalArgumentException` in case of the *newEndpoint* being `null` (and add a throws tag to the JavaDoc).